### PR TITLE
Add examples readme

### PR DESCRIPTION
**Pull request by MTCam**
Add readme to examples dir with minimal detail.  Hopefully help in effort to thin down the examples and improve CI turn-around.
 
cc: @thomasgibson

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+unique features they exercise are as follows:
+
+- autoignition-mpi.py: Chemistry with Pyrometheus
+- heat-source-mpi.py: Diffusion operator
+- lump-mpi.py: Lump advection, advection verification case
+- mixture-mpi.py: Mixture EOS with Pyrometheus
+- scalar-lump-mpi.py: Scalar component lump advection verification case
+- pulse-mpi.py: Acoustic pulse in a box, wall boundary test case
+- sod-mpi.py: Sod's shock case: Fluid test case with strong shock
+- vortex-mpi.py: Isentropic vortex advection: outflow boundaries, verification
```
Comment (inducer) [inline examples/README.md:16]:
```suggestion
- `autoignition-mpi.py`: Chemistry with Pyrometheus
- `heat-source-mpi.py`: Diffusion operator
- `lump-mpi.py`: Lump advection, advection verification case
- `mixture-mpi.py`: Mixture EOS with Pyrometheus
- `scalar-lump-mpi.py`: Scalar component lump advection verification case
- `pulse-mpi.py`: Acoustic pulse in a box, wall boundary test case
- `sod-mpi.py`: Sod's shock case: Fluid test case with strong shock
- `vortex-mpi.py`: Isentropic vortex advection: outflow boundaries, verification
```

Comment (inducer) [reply examples/README.md:16]: Also, maybe it's time to drop the `-mpi` suffix?

**Review by inducer — APPROVED** [review]
This is a good idea, thanks for doing this!

**Comment by matthiasdiener** [comment]
Maybe add wave-eager?

**Comment by MTCam** [comment]
> Maybe add wave-eager?

I don't know what to say about it because it doesn't exercise anything in *MIRGE-Com*?

**Comment by matthiasdiener** [comment]
> > Maybe add wave-eager?
> 
> I don't know what to say about it because it doesn't exercise anything in _MIRGE-Com_?

Fair :-)

**Review by matthiasdiener — APPROVED** [review]

**Pull request merged**